datahub-web-react/src/app/lineage/utils/extendAsyncEntities.ts

```diff
@@ -52,18 +52,42 @@ function extendColumnLineage(
         lineageVizConfig.fineGrainedLineages.forEach((fineGrainedLineage) => {
             fineGrainedLineage.upstreams?.forEach((upstream) => {
                 const [upstreamEntityUrn, upstreamField] = breakFieldUrn(upstream);
-                fineGrainedLineage.downstreams?.forEach((downstream) => {
-                    const downstreamField = breakFieldUrn(downstream)[1];
-                    // fineGrainedLineage always belongs on the downstream urn with upstreams pointing to another entity
-                    // pass in the visualized node's urn and not the urn from the schema field as the downstream urn,
-                    // as they will either be the same or if they are different, it belongs to a "hidden" sibling
+
+                if (lineageVizConfig.type === EntityType.DataJob) {
+                    // draw a line from upstream dataset field to datajob
                     updateFineGrainedMap(
                         fineGrainedMap,
                         upstreamEntityUrn,
                         upstreamField,
                         lineageVizConfig.urn,
-                        downstreamField,
+                        upstreamField,
                     );
+                }
+
+                fineGrainedLineage.downstreams?.forEach((downstream) => {
+                    const [downstreamEntityUrn, downstreamField] = breakFieldUrn(downstream);
+
+                    if (lineageVizConfig.type === EntityType.DataJob) {
```

Comment:
My hope of understanding the lineage code is slowly fading into darkness

Reply:
RIP